### ISSUE TITLE
openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. 

### ISSUE THREAD

**Issue by pradeepdev-1995** [jerryjliu/llama_index#6756]
### Bug Description

I am using the AzureOpenAI llm. Not openai LLM directly.
I given the all azure open ai credentails in the code including azure open ai key . But it still shows the error 

```
openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details.
```

and 
```
tenacity.RetryError: RetryError[<Future at 0x7f99f9ea6ca0 state=finished raised AuthenticationError>] 
```

### Version

llama-index-0.7.1

### Steps to Reproduce

```
from llama_index import VectorStoreIndex, SimpleDirectoryReader,LLMPredictor, ServiceContext
from langchain.llms import AzureOpenAI
import os
import openai

os.environ["OPENAI_API_TYPE"] = "type"
os.environ["OPENAI_API_VERSION"] = "version"
os.environ["OPENAI_API_BASE"] = "api_base"
os.environ["OPENAI_API_KEY"] = "azure_open_ai_key"


llm_predictor = LLMPredictor(llm=AzureOpenAI(aptemperature=0, model_name="model_name"))
service_context = ServiceContext.from_defaults(llm_predictor=llm_predictor)


documents = SimpleDirectoryReader('Data/').load_data()
custom_llm_index = VectorStoreIndex.from_documents(documents,service_context=service_context)
custom_llm_query_engine = custom_llm_index.as_query_engine()
response = custom_llm_query_engine.query("who is this text about?")
print(response)
```

**Comment by avi0gaur** [comment]
Run the Below code It should work.

I faced the same issue and after digging into the llama index code, I realized key access is happing in file level scope rather than function level, which creates a problem if you import first and set the env variable.

This should get corrected anyways you can look at the workaround below.

```
os.environ["OPENAI_API_TYPE"] = "type"
os.environ["OPENAI_API_VERSION"] = "version"
os.environ["OPENAI_API_BASE"] = "api_base"
os.environ["OPENAI_API_KEY"] = "azure_open_ai_key"

from llama_index import VectorStoreIndex, SimpleDirectoryReader,LLMPredictor, ServiceContext
from langchain.llms import AzureOpenAI
import os
import openai

llm_predictor = LLMPredictor(llm=AzureOpenAI(aptemperature=0, model_name="model_name"))
service_context = ServiceContext.from_defaults(llm_predictor=llm_predictor)


documents = SimpleDirectoryReader('Data/').load_data()
custom_llm_index = VectorStoreIndex.from_documents(documents,service_context=service_context)
custom_llm_query_engine = custom_llm_index.as_query_engine()
response = custom_llm_query_engine.query("who is this text about?")
print(response)
```

**Comment by pradeepdev-1995** [comment]
@avi0gaur 
Thanks for the reply.
I tried another method and it also worked. Instead of store the credentials in environment variables, store those in openai arguments such as
```
import openai

openai.api_type = "type"
openai.api_base = "api_base"
openai.api_version = "version"
os.environ["OPENAI_API_KEY"] = "azure_open_ai_key"
openai.api_key = os.getenv("OPENAI_API_KEY")
```
it worked for me

**Comment by logan-markewich** [comment]
I will also note that we just released our own Azure LLM implementation (in v0.7.2), which has nicer errors to help you with the setup.

See details here https://gpt-index.readthedocs.io/en/latest/examples/customization/llms/AzureOpenAI.html
